l5kit/l5kit/simulation/utils.py

```diff
@@ -74,3 +74,50 @@ def disable_agents(dataset: ChunkedDataset, allowlist: np.ndarray) -> None:
     dataset.agents["yaw"][mask_disable] *= 0
     dataset.agents["extent"][mask_disable] *= 0
     dataset.agents["label_probabilities"][mask_disable] = -1
+
+
+def get_frames_subset(dataset: ChunkedDataset, frame_start_idx: int, frame_end_idx: int) -> ChunkedDataset:
+    """Get a new dataset with frames between start (included) and end (excluded).
+    Assumptions:
+        the dataset has only 1 scene
+        the dataset is in numpy format and not zarr anymore
+
+    :param dataset: the single-scene dataset.
+    :param frame_start_idx: first frame to keep.
+    :param frame_end_idx: where to stop taking frames (excluded).
+
+    """
+    if not len(dataset.scenes) == 1:
+        raise ValueError(f"dataset should have a single scene, got {len(dataset.scenes)}")
+    if not isinstance(dataset.agents, np.ndarray):
+        raise ValueError("dataset agents should be an editable np array")
+    if not isinstance(dataset.tl_faces, np.ndarray):
+        raise ValueError("dataset tls should be an editable np array")
+    if not isinstance(dataset.frames, np.ndarray):
+        raise ValueError("dataset frames should be an editable np array")
```

Comment:
I would remove that for two reasons: the NumPy array can be non-writeable and we are not checking this here, and if a different type is provided we will see clear errors. Also, if we want to do runtime type-checking, I would go with another solution that uses the annotations for that automatically. So I would just remove the check for the type.

Reply:
> I would remove that for two reasons: the NumPy array can be non-writeable and we are not checking this here

but this doesn't happen in this codebase. You would really need to go behind our API to get a non writable numpy array

> Also, if we want to do runtime type-checking, I would go with another solution that uses the annotations for that automatically. So I would just remove the check for the type.

The only way I see is another class similar to `ChunkedDataset` with numpy arrays. But that would affect a lot of code and it's not clear to me how it would interact with the `EgoDataset`..

Reply:
I see, let's leave the checks then, but when that would be the case that we don't have for instance a numpy array for the `dataset.tl_faces` ?

Reply:
the default `ChunkedDataset` uses `zarr.arrays` when there is a map with the disk. We have `numpy.array` only after calling functions such as `get_scene_dataset` which get a memory only version of a subset of the original zarr

Reply:
Ok now I see the problem, it really makes sense to have another dataset or a flag in the dataset saying when it is in-memory or persistent data. But that doesn't make sense to do in this PR. Thanks !